nav2_costmap_2d/test/unit/collision_footprint_test.cpp

```diff
@@ -0,0 +1,94 @@
+#include <string>
+#include <vector>
+#include <memory>
+
+#include "gtest/gtest.h"
+#include "nav2_costmap_2d/collision_footprint.hpp"
+
+TEST(collision_footprint, footprintCostWithPose)
+{
+  std::shared_ptr<nav2_costmap_2d::Costmap2D> costmap_ =
+    std::make_shared<nav2_costmap_2d::Costmap2D>(100, 100, 0.1, 0, 0, 0);
+
+  geometry_msgs::msg::Point p1;
+  p1.x = -0.5;
+  p1.y = 0.0;
+  geometry_msgs::msg::Point p2;
+  p2.x = 0.0;
+  p2.y = 0.5;
+  geometry_msgs::msg::Point p3;
+  p3.x = 0.5;
+  p3.y = 0.0;
+  geometry_msgs::msg::Point p4;
+  p4.x = 0.0;
+  p4.y = -0.5;
+
+  nav2_costmap_2d::Footprint footprint = {p1, p2, p3, p4};
+
+  nav2_costmap_2d::CollisionFootprint collision_footprint(costmap_);
+
+  auto value = collision_footprint.footprintCostWithPose(5.0, 5.0, 0.0, footprint);
+
+  EXPECT_NEAR(value, 0.0, 0.001);
```

Comment:
Shouldn't this work with 0.0 and not near? This is deterministic

Reply:
I can do that but I have this OCD with comparing floating types. So, as a measure I use near.

Reply:
Sure - we can keep it